internal/mode/static/telemetry/collector_test.go

```diff
@@ -127,21 +140,42 @@ var _ = Describe("Collector", Ordered, func() {
 			Version:             version,
 			PodNSName:           podNSName,
 		})
-		k8sClientReader.GetCalls(createGetCallsFunc(ngfPod, ngfReplicaSet))
+
+		baseGetCalls = createGetCallsFunc(ngfPod, ngfReplicaSet, kubeNamespace)
+		k8sClientReader.GetCalls(baseGetCalls)
 	})
 
+	mergeGetCallsWithBase := func(f getCallsFunc) getCallsFunc {
+		return func(
+			ctx context.Context,
+			nsName types.NamespacedName,
+			object client.Object, option ...client.GetOption,
+		) error {
+			err := baseGetCalls(ctx, nsName, object, option...)
+			Expect(err).ToNot(HaveOccurred())
+
+			return f(ctx, nsName, object, option...)
+		}
+	}
+
 	Describe("Normal case", func() {
 		When("collecting telemetry data", func() {
 			It("collects all fields", func() {
 				nodes := []v1.Node{
 					{
-						ObjectMeta: metav1.ObjectMeta{Name: "node1"},
+						ObjectMeta: metav1.ObjectMeta{
```

Comment:
Is there a happy path test case for cluster ID? I don't see the clusterID included in the expected data in this test...

Reply:
Yes, its right here - https://github.com/nginxinc/nginx-gateway-fabric/blob/88b9b0cdd983df43f66c31203bff5e46985f0728/internal/mode/static/telemetry/collector_test.go#L126